app/uk.gov.hmrc.helptosaveproxy/auth/Auth.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosaveproxy.auth
+
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import uk.gov.hmrc.auth.core.AuthProvider.{GovernmentGateway, PrivilegedApplication}
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthProviders}
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.helptosaveproxy.util.{Logging, WithMdcExecutionContext}
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
+
+import scala.concurrent.Future
+
+class Auth(val authConnector: AuthConnector) extends AuthorisedFunctions { this: BaseController with Logging with WithMdcExecutionContext ⇒
+
+  val authProviders: AuthProviders = AuthProviders(GovernmentGateway, PrivilegedApplication)
+
+  type HtsAction = Request[AnyContent] ⇒ Future[Result]
+
+  def authorised(action: HtsAction): Action[AnyContent] =
+    Action.async { implicit request ⇒
+      authorised(authProviders) {
+        action(request)
+      }.recover { handleFailure() }
+    }
+
+  def handleFailure(): PartialFunction[Throwable, Result] = {
+    case e: NoActiveSession ⇒
+      logger.warn(s"no active session was found for api user, reason: ${e.reason}")
```

Comment:
it may not be `api` request always -  so please adjust the logging with some generic message ?